gk/main.c

```diff
@@ -196,16 +213,24 @@ drop_packet(struct rte_mbuf *pkt)
  * (3) put this encapsulated packet in the request queue.
  */
 static int
-gk_process_request(struct flow_entry *fe, struct ipacket *packet)
+gk_process_request(struct flow_entry *fe,
+	struct ipacket *packet, struct gk_lpm *ltbl)
 {
 	int ret;
 	uint64_t now = rte_rdtsc();
 	uint8_t priority = priority_from_delta_time(now,
 			fe->u.request.last_packet_seen_at);
-
-	/* TODO The tunnel information should come from the LPM table. */
-	struct ipip_tunnel_info tunnel;
-	memset(&tunnel, 0, sizeof(tunnel));
+	struct gk_policy *policy =  &ltbl->policy_tbl[fe->grantor_id];
+	struct ipip_tunnel_info *tunnel;
+	bool stale = true;
+
+	if (policy->action == GK_FWD_GATEWAY) {
```

Comment:
If the code reaches here and `policy->action != GK_FWD_GRANTOR`, it's a bug.

The only way to call `gk_process_request()` is that the LPM table identifies the destination as a destination that must go through a grantor server.